crates/ty_python_semantic/src/types/display.rs

```diff
@@ -1660,6 +1674,17 @@ impl<'db> FmtDetailed<'db> for DisplaySignature<'_, 'db> {
         // When we exit this function, write a marker signaling we're ending a signature
         let mut f = f.with_detail(TypeDetail::SignatureEnd);
 
+        // If we're multiline printing and a name hasn't been emitted, try to
+        // make one up to make things more pretty
+        if self.settings.multiline
+            && !self.settings.disallow_signature_name
+            && let Some(definition) = self.definition
+            && let Some(name) = definition.name(self.db)
```

Comment:
This seems reasonable, if we're tracking a definition for the callable, that name should be a reasonable one to use.

That said, if this is all we're doing it makes me wonder if/why we even need `self.settings.disallow_signature_name` and threading down the definition, for this. Presumably we already had access to the definition in whatever top-level type we were initially rendering that contains a signature; why not always render it there?